src/logic/factory-logic.ts

```diff
@@ -35,6 +35,13 @@ import { SubClass } from '../models/subclass';
 import { Title } from '../models/title';
 import { Utils } from '../utils/utils';
 
+type RequiredAbilityProps = 'id' | 'name' | 'type' | 'distance' | 'target';
+type OptionalAbilityProps = 'description' | 'keywords' | 'cost' | 'signature' | 'preEffect' | 'powerRoll' | 'effect' | 'strained' | 'alternateEffects';
+interface InitAbility extends Pick<Ability, RequiredAbilityProps>, Partial<Pick<Ability, OptionalAbilityProps>> {
+	spend?: { value?: number, effect: string }[];
+	persistence?: { value?: number, effect: string }[];
+}
```

Comment:
I find this quite hard to read, and I don't understand how it's better than what we had before.

Reply:
I think we can get to a type definition where we can specify which props of the model are optional for the factory, rather than duplicating the model definition for the factory. I'll revert it for this PR, and workshop it a bit more for a separate PR.